bucket/bucket_plain.go

```diff
@@ -5,7 +5,49 @@ import (
 )
 
 // MetricOperation is a list of metric operations to use for metric
-type MetricOperation [3]string
+type MetricOperation struct {
+	operations []string
+	Labels     map[string]string
+}
+
+// NewMetricOperation  builds and returns new MetricOperation instance with defined label keys
+func NewMetricOperation(operations ...string) MetricOperation {
+	ops := []string{"-", "-", "-"}
+
+	for i := 0; i < 3; i++ {
+		if len(operations) <= i {
```

Comment:
it would be better to do something like:
```
switch opsLen := len(operations):
case opsLen == 3: (just assign)
case opsLen > 3: (get the slice)
case opsLen < 3: (do the loop)
```